src/api/application-discovery-service.js

```diff
@@ -0,0 +1,129 @@
+const BASE_PATH = '/ads';
+
+function mixin(client) {
+  client.ads = {
```

Comment:
There's one catch if we do it this way: We lose the benefit of IntelliSense. It would be much nicer if we could type `whateverClient.` and then the code editor could suggest every avaiable method with the information about how to use it.